packages/itwin/grouping-mapping-widget/src/components/Mappings/hooks/useRunExtraction.ts

```diff
@@ -25,12 +25,13 @@ export const useRunExtraction = ({
     mutationKey: ["runExtraction"],
     mutationFn: async (mappings: Mapping[]) => {
       const accessToken = await getAccessToken();
-      const mappingIds: ExtractionRequestMapping[] = mappings.length > 0 ? mappings.map((mapping) => { return { id: mapping.id }; }) : [];
-      const extractionRequest: ExtractionRunRequest = {
+      const mappingIds: ExtractionMapping[] = mappings.length > 0 ? mappings.map((mapping) => { return { id: mapping.id }; }) : [];
```

Comment:
Providing an empty `mappings` array will result in 422 response. This should not default to an empty array. If there are no mappings selected for extraction, the `runExtraction` should not even be called. In V2 the `mappings` property for running extraction is required and it must not be empty.

Reply:
In `MappingsView.tsx` the extraction button is disable if `selectedMappings.length === 0` , `runExtraction` is not used if so.

Reply:
Conditional added.

Reply:
Resolved.